proposals/module-types/Explainer.md

```diff
@@ -548,37 +520,31 @@ As syntactic sugar in the textual format, a nested module can simply use the
   )
 )
 ```
-The desugared form of `$child` would explicitly create the alias with a `type`
-definition which references the parent:
+This gets desugared into an explicit `alias` definition adding an entry to
+the child's type index space:
 ```wasm
 (module $Parent
   (type $WasiFile (instance $wasi-file
     (export "read" (func (param i32 i32 i32) (result i32)))
   ))
   (module $child
-    (type $WasiFile (alias $Parent.$WasiFile))
+    (type $WasiFile (alias $Parent $WasiFile))
```

Comment:
I don't follow: the `$Parent` module has its own `$WasiFile` typedef, and thus the `$Parent` can use that for declaring the module type of `$child`.  The `alias` is necessary to inject `$WasiFile` into the type section of `$child` for the import definition in `$child`'s Import Section.

Reply:
Ah, right, I was thinking of a case where this is a real projection.

Rethinking what I meant, I realised that it is yet another case. Namely, what about projections local to a module _type_? For example, how would we represent the type of the following module:
```
(module
  (instance $I (import "I") (instance (type $T (export "T") ...))))
  (type $T (alias $I $T))
  (func (export "f") (param (ref $T)))
)
```
Sections don't help with this, since all those bindings are local to a module type. So module (and instance) types need to have a way for defining local type aliases, too. But they can already contain type definitions, so this should be a natural extensions.

Reply:
Yep, that makes sense.  Even ignoring the instance type and projection, I think we'll need the ability to locally reference a type import (and then, in the associated instance type, some sort of free-floating `(type $T)`).  But, iiuc, there's nothing to change in the current proposal text?

Reply:
Yeah, no change needed right now, but at some point the proposal will need to include an explanation of how intra-type scoping is gonna work and how it is encoded.

Reply:
Yes, makes sense; thanks for pointing it out.